Source/WebCore/workers/service/ServiceWorkerRegistration.cpp

```diff
@@ -311,7 +311,21 @@ void ServiceWorkerRegistration::showNotification(ScriptExecutionContext& context
         return;
     }
 
-    if (auto* serviceWorkerGlobalScope = dynamicDowncast<ServiceWorkerGlobalScope>(context)) {
+    auto* serviceWorkerGlobalScope = dynamicDowncast<ServiceWorkerGlobalScope>(context);
+
+    // If we're handling a PushNotificationEvent, this Notification will override the proposed notification
+    // instead of being shown directly.
+#if ENABLE(DECLARATIVE_WEB_PUSH)
+    if (serviceWorkerGlobalScope) {
+        if (auto* pushNotificationEvent = serviceWorkerGlobalScope->pushNotificationEvent()) {
```

Comment:
RefPtr

Reply:
Not needed? It's not stored for later use and cannot possibly be cleared here before this use is done.

Reply:
Ryosuke and I are fixing them after you land code like this because the static analyzer complains. You're calling a function on it on the next line so you need to hold a strong ref to it. Recent policy change for safety.